football_dictionaries/assignment_2.py

```diff
@@ -1,2 +1,18 @@
+"""
+# code from
+#https://stackoverflow.com/questions/31071888/python-group-list-items-in-a-dict
+
+#Working / tested and passed.
+"""
+
+
 def players_by_position(squads_list):
-    pass
+    groupby_dict = {}
+    new_squads_data = [
+        {'number': x[0], 'position': x[1], 'name': x[2], 'date_of_birth': x[3], 'caps': x[4], 'club': x[5],
```

Comment:
Hey D,

I think the purpose of the exercise was the create a module and import it. instead of recreating the new_squads_data. I see you've done that in assignement 3.  

And you can use the setdefault method to create a new dictionary from the new_squats_data dictionary. 

looks like this : 
for player in squad:
        position = player['position']  #"GK" 
        players_by_position_dict.setdefault(position, []) #{GK: [{}]}
        players_by_position_dict[position].append(player)